src/extensions/segment-integration.ts

```diff
@@ -34,23 +47,26 @@ interface SegmentPlugin {
     version: string
     type: 'enrichment'
     isLoaded: () => boolean
-    load: (ctx: SegmentPluginContext, instance: any, config?: any) => Promise<unknown>
-    unload?: (ctx: SegmentPluginContext, instance: any) => Promise<unknown> | unknown
+    load: (ctx: SegmentContext, instance: any, config?: any) => Promise<unknown>
+    unload?: (ctx: SegmentContext, instance: any) => Promise<unknown> | unknown
     ready?: () => Promise<unknown>
-    track?: (ctx: SegmentPluginContext) => Promise<SegmentPluginContext> | SegmentPluginContext
-    identify?: (ctx: SegmentPluginContext) => Promise<SegmentPluginContext> | SegmentPluginContext
-    page?: (ctx: SegmentPluginContext) => Promise<SegmentPluginContext> | SegmentPluginContext
-    group?: (ctx: SegmentPluginContext) => Promise<SegmentPluginContext> | SegmentPluginContext
-    alias?: (ctx: SegmentPluginContext) => Promise<SegmentPluginContext> | SegmentPluginContext
-    screen?: (ctx: SegmentPluginContext) => Promise<SegmentPluginContext> | SegmentPluginContext
+    track?: (ctx: SegmentContext) => Promise<SegmentContext> | SegmentContext
```

Comment:
nit that w could extract an interface here and this block of repeating lines would be easier on the eyes